addons/web_editor/static/tests/link_test.js

```diff
@@ -0,0 +1,458 @@
+                );
+            }
+        );
+
+        QUnit.test("typing www.odoo + space should not convert to link", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            inputText(".odoo-editor-editable p", "www.odoo");
+            insertText(editor, " ");
+            editor.clean();
+            assert.strictEqual(editable.innerHTML, "<p>www.odoo </p>");
+        });
+
+
+        QUnit.module("Toolbar based");
+
+        QUnit.test("should convert all selected text to link", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            insertText(editor, "Hello");
+            triggerEvent(node, "keydown", { key: "a", ctrlKey: true });
+            await nextTick();
+            editor.document.querySelector(".fa-link").click();
+            await nextTick();
+            inputText('input[id="o_link_dialog_url_input"]', "#");
+            editor.document.querySelector(".o_dialog footer button.btn-primary").click();
+            await nextTick();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="#" target="_blank">Hello</a></p>`
+            );
+        });
+
+        QUnit.test("should set the link on two existing characters", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            insertText(editor, "Hello");
+            setSelection(node, 1, node, 3);
+            editor.document.querySelector(".fa-link").click();
+            await nextTick();
+            inputText('input[id="o_link_dialog_url_input"]', "#");
+            editor.document.querySelector(".o_dialog footer button.btn-primary").click();
+            await nextTick();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p>H<a href="#" target="_blank">el</a>lo</p>`
+            );
+        });
+
+        QUnit.test("Should be able to insert link on empty p", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            triggerEvent(node, "keydown", { key: "a", ctrlKey: true });
+            await nextTick();
+            editor.document.querySelector(".fa-link").click();
+            await nextTick();
+            inputText('input[id="o_link_dialog_url_input"]', "#");
+            editor.document.querySelector(".o_dialog footer button.btn-primary").click();
+            await nextTick();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="#" target="_blank"><br></a></p>`
+            );
+        });
```

Comment:
This scenario is kinda weird. I think inserting a link on a empty P is best done via /link (powerbox), or ctrl+k ctrl+k (which is already tested).

Reply:
Since it was a possibility for adding a link , hence i added it. So should i remove this?